### DISCUSSION TITLE
Bug: macOS: background flickers at 1/2 or 1/4 size when creating new tabs/splits

### DISCUSSION THREAD

**Discussion by kdrag0n** [ghostty-org/ghostty#4180] [Feature Requests, Ideas] [answered]
Steps to reproduce:
- Press Cmd-T

Config: none

New tabs/splits are momentarily only occupying 1/2 or 1/4 of their full rectangle, with the rest of the background being transparent so the app behind it bleeds through.

I think this is a regression between 2fb92dd4 (good) and ecfca17a (bad).

https://github.com/user-attachments/assets/ab49439a-a26c-4505-ba98-9b496d60f62d

**Comment by mitchellh** [comment]
I'm pretty sure this is also a dupe, but can't find that one right now.

**Reply by jwadolowski** [reply]
I've noticed the same thing on my machine - it is clearly visible on `new_tab` (underlying windows/wallpaper shows up for a split second) and `move_tab` (the other tab flickers upon movement).

macOS Sequoia 15.2
Ghostty 1.0.1

That's most likely a duplicate of #4181

https://github.com/user-attachments/assets/3c0e17b5-7359-4945-b7b7-f5ad57758984

https://github.com/user-attachments/assets/934e8024-5abe-4ab9-95a8-7f12ab018a0a

**Reply by injust** [reply]
I think the `move_tab` behaviour is a separate issue. Notably, what briefly appears is the _tab_ that is underneath (as in the tab you'd see if you dragged the current tab out of the window), not the wallpaper or another window.

**Reply by jwadolowski** [reply]
I've just created a separate discussion for the `move_tab` issue: #5651

**Comment by maruks** [comment]
please fix this bug

**Comment by injust** [comment]
This feels similar to what I reported at https://discord.com/channels/1005603569187160125/1005603569711452192/1272554298571493498, but that was months ago, so it can't be the same issue if the listed good/bad commits are correct.

**Comment by mitchellh** [comment, marked as answer]
This is fixed on tip.
